main.go

```diff
@@ -76,17 +79,38 @@ func main() {
 		os.Exit(1)
 	}
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Namespace:          namespace,
-		Scheme:             scheme,
-		MetricsBindAddress: metricsAddr,
-		Port:               9443,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "988b4062.3scale.net",
-	})
-	if err != nil {
-		setupLog.Error(err, "unable init the manager")
-		os.Exit(1)
+	var mgr ctrl.Manager
+	// If a watch namespace is detected (i.e. operator is namespace scoped), then pass the NS to cache.Options.DefaultNamespaces
+	// If no watch namespace is detected (i.e. operator is cluster scoped), then don't specify cache.Options.DefaultNamespaces
+	if namespace != "" {
+		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
```

Comment:
what about one single `mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{...}`

And have `Cache` object with some value or another depending of the namespace scoping?

Reply:
I'll look into this but I think the issue we'll run into is we'll be passing in a nil `Cache` object when it's cluster scoped which could cause problems. The `Cache.DefaultNamespaces` field needs to be unset for cluster scoped.

Reply:
Actually it looks like you can pass `cache.AllNamespaces` which would fix the nil `cache` object issue, I'll push this change as a new commit

Reply:
This is more for future readers but I was unable to use the `cache.AllNamespaces` string as intended. So I changed the implementation to still use one single `mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{...}` but instead have a `Cache` object which takes the namespace if it exists and otherwise is passed as an empty object.